app/scripts/lib/fxa-client.js

```diff
@@ -132,6 +132,8 @@ define(function (require, exports, module) {
       }
 
       var updatedSessionData = {
+        challengeMethod: accountData.challengeMethod,
```

Comment:
I know it's a bit late in the day to be suggesting this but inconsistent names have been a bug-bear for me in the past. Anyway, given the name of the feature is "sign-in confirmation" and we're already trying to avoid referring to "verification" because of that, should these response fields from the auth server be instead be called `confirmationMethod` and `confirmationReason`?

I know we agreed on `challenge...` already but maybe there's value in being consistent across the board.